dask/dataframe/rolling.py

```diff
@@ -111,9 +114,12 @@ def _call_method(self, method_name, *args, **kwargs):
         if self.rolling_kwargs['axis'] in [0, 'rows']:
             # roll in the partition direction (will need to access neighbor)
             for i in range(1, self.obj.npartitions + 1):
+                # Get just the needed values from the previous partition
+                window = self.rolling_kwargs['window']
+                prior_chunk_tail = (tail, (old_name, i-1), window-1)
```

Comment:
We will need to pull out the tail operation to a separate task.

```python
tail_name = name + '-tail-' + tokenize(...)
...
for ...
    ...
    dsk[tail_name, i] = (tail, (old_name, i-1), window-1)
    dsk[new_name, i] = (call_pandas_rolling_method_with_neighbor, (tail_name, i), (old_name, i))
```

The scheduler keeps track of each value in the task graph as an atomic datum.  In order for the tail to be separable for communication we need to make it a separate key-value pair in the graph.